src/gpio.rs

```diff
@@ -260,6 +259,33 @@ where
         }
     }
 
+    /// Set pin direction to dynamic (i.e. changeable at runtime)
+    ///
+    /// This method is only available while the pin is in input mode.
```

Comment:
Technically this is correct, but also a bit confusing since an identical method is available in input mode too. Maybe say that this method is available, unless the pin is already in dynamic mode?

It should also be possible to only write this method once, by defining a `direction::NotDynamic` trait and using `where Direction: NotDynamic`. No need to do this now though.

Reply:
Ah, yeah that makes sense. I've corrected the comment, but I'm afraid I don't quite get how your `direction::NotDynamic` design would work: The duplicate `into_dynamic()` functions already are part of a `direction::[Input |  Output]` impl, how can two direction traits apply at the same time?

Reply:
The basic idea is that `direction::NotDynamic` is implemented for `direction::[Input | Output]` (i.e. one implementation for `direction::Input` & another one for `direction::Output`, not both at the same time) and `into_dynamic()` is only implemented with `where Direction: NotDynamic`, not for `direction::[Input | Output]` directly. We've already used this pattern [in the past](https://github.com/lpc-rs/lpc8xx-hal/blob/688aba523b12f34559623252959cc618d6d2bf12/src/gpio.rs#L328), but I think we dropped the `Unknown` state & removed it. Hope that helps

Reply:
oh, okay! Thank you for the explanation. If that's okay with you I'd postpone this to a later PR; I've written down a TODO for myself about this though.